airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -95,42 +99,64 @@ def parse_response(self, response: SearchPager) -> Iterable[Mapping]:
             yield self.google_ads_client.parse_single_result(self.get_json_schema(), result)
 
     def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
-        for customer_id in self.google_ads_client.customer_ids:
-            self._customer_id = customer_id
-            yield {}
+        for customer in self.customers:
+            yield {"customer_id": customer.id}
 
     def read_records(self, sync_mode, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
-        if not stream_slice:
+        if stream_slice is None:
```

Comment:
why this change? when would a slice ever be none?

Reply:
when the input state contains abnormally large value - the slice is None and no records are emitted